modules/lib/core/index.d.ts

```diff
@@ -0,0 +1,79 @@
+declare global {
+    interface XpXData {
+    }
+}
+
+export type UserKey = `user:${string}:${string}`;
+export type GroupKey = `group:${string}:${string}`;
+export type RoleKey = `role:${string}`;
+
+export type PrincipalKey = UserKey | GroupKey | RoleKey;
+
+export type WorkflowState = 'IN_PROGRESS' | 'PENDING_APPROVAL' | 'REJECTED' | 'READY';
+
+export type WorkflowCheckState = 'PENDING' | 'REJECTED' | 'APPROVED';
+
+export type ContentInheritType = 'CONTENT' | 'PARENT' | 'NAME' | 'SORT';
```

Comment:
I would consider inlining these straight in the interfaces that uses them because: 
 1. I don't think anybody will use these types in application code.
 2. They will be located where they are used without abstraction (you don't have to jump around in the code)
 3. They can always be derived in application if somebody absolutely need them:
     ```typescript
     type WorkflowState = Workflow['state'];
     type ContentInheritType = NonNullable<Content['inherit']>
     ```